pallets/ema-oracle/src/lib.rs

```diff
@@ -387,11 +387,15 @@ impl<T: Config> OnTradeHandler<AssetId, Balance> for OnActivityHandler<T> {
 		liquidity_a: Balance,
 		liquidity_b: Balance,
 	) -> Result<Weight, (Weight, DispatchError)> {
-		// We assume that zero values are not valid and can be ignored.
-		if liquidity_a.is_zero() || liquidity_b.is_zero() || amount_a.is_zero() || amount_b.is_zero() {
-			log::warn!(target: LOG_TARGET, "Neither liquidity nor amounts should be zero. Source: {source:?}, liquidity: ({liquidity_a},{liquidity_b}), amounts: {amount_a}/{amount_b}");
+		// We assume that zero liquidity values are not valid and can be ignored.
+		if liquidity_a.is_zero() || liquidity_b.is_zero() {
+			log::warn!(
+				target: LOG_TARGET,
+				"Liquidity amounts should not be zero. Source: {source:?}, liquidity: ({liquidity_a},{liquidity_b})"
+			);
 			return Err((Self::on_trade_weight(), Error::<T>::OnTradeValueZero.into()));
 		}
+
 		let price = determine_normalized_price(asset_a, asset_b, liquidity_a, liquidity_b);
```

Comment:
@apopiak @enthusiastmartin i dont think this works for stableswap as it just divides reserves to determine price, this works only for xyk and is not the way how stableswap determines spot price (neither LBP)

Reply:
i think stableswap or LBP or any other amm than XYK has to be able to set price directly

Reply:
https://github.com/galacticcouncil/HydraDX-simulations/blob/01d560611efef167988247579eb897cfc505c47d/hydradx/model/amm/stableswap_amm.py#L111

Reply:
You'll need a param. This can still be used if you think of it as nom/denom instead (of the result of the equation to get the price) it should be renamed then probably